python/tests/test_highlevel.py

```diff
@@ -1660,6 +1661,36 @@ def test_newick(self):
             for tree in ts.trees():
                 self.verify_newick(tree)
 
+    def test_as_dict_of_dicts(self):
+        for ts in get_example_tree_sequences():
+            tree = next(ts.trees())
+            print(tree)
+
+            adj_dod = tree.as_dict_of_dicts()
+            print(adj_dod)
+
+            g = nx.DiGraph(adj_dod)
+            print(g.nodes)
+
+            self.verify_nx_graph_topology(tree, g)
+
+    def verify_nx_graph_topology(self, tree, g):
+        self.assertSetEqual(set(tree.nodes()), set(g.nodes))
+
+        roots = {n for n in g.nodes if g.in_degree(n) == 0}
+        self.assertSetEqual(set(tree.roots), roots)
+
+        leaves = {n for n in g.nodes if g.out_degree(n) == 0}
+        self.assertSetEqual(set(tree.leaves()), leaves)
+
+        for root in tree.roots:
+            print(root)
+            self.assertSetEqual(
+                set(u for u in tree.nodes() if tree.is_descendant(u, root) and
+                    u != root),
```

Comment:
Is it a bug that a sole root node is also considered a descendant by `Tree.is_descendant`?